src/ws/cockpithandlers.c

```diff
@@ -720,3 +721,36 @@ cockpit_handler_ping (CockpitWebServer *server,
 
   return TRUE;
 }
+
+gboolean
+cockpit_handler_ca_cert (CockpitWebServer *server,
```

Comment:
Can you please add an integration test in check-connection to this? Something like `curl /ca.cer | openssl x509 -in - -text`? `testTls` also covers a custom certificate, you could extend that to test the 404 case.

Reply:
Done, with a fake certificate.